apps/api/src/common/authority-checker.service.ts

```diff
@@ -0,0 +1,380 @@
+import { PrismaClient, Authority, Workspace } from '@prisma/client'
+import { VariableWithProjectAndVersion } from 'src/variable/variable.types'
+import {
+  BadRequestException,
+  Injectable,
+  NotFoundException,
+  UnauthorizedException
+} from '@nestjs/common'
+import getCollectiveProjectAuthorities from './get-collective-project-authorities'
+import getCollectiveWorkspaceAuthorities from './get-collective-workspace-authorities'
+import { EnvironmentWithProject } from 'src/environment/environment.types'
+import { ProjectWithSecrets } from 'src/project/project.types'
+import { SecretWithProjectAndVersion } from 'src/secret/secret.types'
+
+export interface AuthorityInput {
+  userId: string
+  entity: {
+    id?: string
+    name?: string
+  }
+  authority: Authority
+  prisma: PrismaClient
+}
+
+@Injectable()
+export class AuthorityCheckerService {
+  public async checkAuthorityOverWorkspace(
+    input: AuthorityInput
+  ): Promise<Workspace> {
+    const { userId, entity, authority, prisma } = input
+
+    let workspace: Workspace
+
+    try {
+      if (entity?.id) {
+        workspace = await prisma.workspace.findUnique({
+          where: {
+            id: entity.id
+          }
+        })
+      } else {
+        workspace = await prisma.workspace.findFirst({
+          where: {
+            name: entity?.name
+          }
+        })
+      }
+    } catch (error) {
+      /* empty */
```

Comment:
Use the Logger class to log out the error details in here. Additionally, re-throw the error